test/functional/api/ui_spec.lua

```diff
@@ -34,4 +35,13 @@ describe('nvim_ui_attach()', function()
     expect_err('UI already attached to channel: 1',
                request, 'nvim_ui_attach', 40, 10, { rgb=false })
   end)
+  it('autocmds UIAttach/Detach set v:event', function()
+    local screen = Screen.new()
+    command('autocmd UIAttach * :let g:ui_attach_v_event_has_channel_id = has_key(v:event, "channel_id")')
```

Comment:
you can store the actual channel id, we know it to be `1` for the stdio channel used by tests.

Reply:
Changed this to compare the full dict